openedx/core/djangoapps/schedules/signals.py

```diff
@@ -41,17 +41,20 @@ def create_schedule(sender, **kwargs):
         log.debug('Schedules: Creation not enabled for this course or for this site')
         return
 
+    if not enrollment.course_overview.self_paced:
```

Comment:
Nit Perf suggestion: does the call to `enrollment.course_overview` make a SQL query?  If not, consider moving this check up before the call to `get_current_site`.  This way, we don't make the additional queries (for getting the site and for checking schedule-configs) for each enrollment creation.